### ISSUE TITLE
Modal Dialogs not working

### ISSUE THREAD

**Issue by braxtonmckee** [APrioriInvestments/typed_python#171]
I get

```
Modal.js:46 Uncaught ReferenceError: mainStyle is not defined
    at Modal.render (Modal.js:46)
    at CellHandler.handleMessage (CellHandler.js:181)
    at CellSocket.handleRawMessage (CellSocket.js:239)
render @ Modal.js:46
handleMessage @ CellHandler.js:181
handleRawMessage @ CellSocket.js:239
```

in the main ui. We should add a playground thing for this.

**Comment by darth-cheney** [comment]
I'm not sure I've actually seen one of these modals in action before. Can you shoot me over an example (or a quick playground) when you get the chance?
  
Error message you're seeing makes it look like an easy fix, but you never know

**Comment by braxtonmckee** [comment]
see 2fcf9c948c4cbd6b23349e2e1912d71176645bb1

**Comment by darth-cheney** [comment]
Hmm I am not able to see anything in this example -- not even the Card is rendering. I'm able to fix that by wrapping the final statement in return in parens (so that the whole method doesn't return None), but then there seem to be other data/Slot issues here that prevent the Modal from even trying to render.

**Comment by braxtonmckee** [comment]
maybe try modifying 'cells' to not show the modal conditionally and make
sure that works first?

**Comment by darth-cheney** [comment]
Yeah I have a working version of a basic Modal, but it's not completely "right". Had this ever worked in the past? Do you mind if I change up the API a bit? It seems to me we want a Slot on whether the modal is being "shown" or not, and that all modals should at least initially render hidden, and then "show" if they need to

**Comment by braxtonmckee** [comment]
yes this did work in the past. But I just hacked it together. Please do
make it better. Ideally we actually only ever have 1 modal dialog being
shown, and we use it for things like text edit boxes or rename boxes, etc.

**Comment by braxtonmckee** [comment]
might make sense to create something like a 'ModalContainer' and then have
the 'Modal' look up the tree to find the container it's in, so that if you
have two sub-ui's with modals you can actually model that?

**Comment by darth-cheney** [comment]
Closed by #196
